keps/4377-metrics-tls/README.md

```diff
@@ -0,0 +1,337 @@
+[Recommended For Production](https://book.kubebuilder.io/reference/metrics#recommended-enabling-certificates-for-production-disabled-by-default)
+
+> The default scaffold in cmd/main.go uses a controller-runtime feature to automatically generate a self-signed certificate to secure the metrics server.
+  While this is convenient for development and testing, it is not recommended for production.
+  Those certificates are used to secure the transport layer (TLS). The token authentication using authn/authz, which is enabled by default serves as the application-level credential. However, for example, when you enable the integration of your metrics with Prometheus, those certificates can be used to secure the communication.
+
+With the deprecation of Kube-RBAC-Proxy from Kueue, we are already requiring secure metrics via the use of the below code:
+
+```golang
+if secureMetrics {
+  ...
+  metricsServerOptions.FilterProvider = filters.WithAuthenticationAndAuthorization
+}
+```
+
+### Prometheus Service Monitor Changes
+
+The example below shows:
+
+```yaml
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: controller-manager-metrics-monitor
+  namespace: system
+spec:
+  endpoints:
+    - path: /metrics
+      port: https
+      scheme: https
+      bearerTokenFile: /var/run/secrets/kubernetes.io/serviceaccount/token
+      tlsConfig:
+        serverName: SERVICE_NAME.SERVICE_NAMESPACE.svc
+        insecureSkipVerify: false
+        ca:
+          secret:
+            name: metrics-server-cert
+            key: ca.crt
+        cert:
+          secret:
+            name: metrics-server-cert
+            key: tls.crt
+        keySecret:
+          name: metrics-server-cert
+          key: tls.key
+  selector:
+    matchLabels:
+      app.kubernetes.io/component: controller
+      app.kubernetes.io/name: kueue
+```
+
+### Deployment Changes
+
+Whenever you add Certificate management, one needs to mount a volume to the deployment that references a secret that is created by
+an external service.
+
+Following the kubebuilder examples, one would add the following example to the deployment:
+
+```yaml
+# Add the volumeMount for the metrics-server certs
+- op: add
+  path: /spec/template/spec/containers/0/volumeMounts/-
+  value:
+    mountPath: /tmp/k8s-metrics-server/metrics-certs
+    name: metrics-certs
+    readOnly: true
+
+# Add the metrics-server certs volume configuration
+- op: add
+  path: /spec/template/spec/volumes/-
+  value:
+    name: metrics-certs
+    secret:
+      secretName: metrics-server-cert
+      optional: false
+      items:
+        - key: ca.crt
+          path: ca.crt
+        - key: tls.crt
+          path: tls.crt
+        - key: tls.key
+          path: tls.key
+```
+
+### Controller Changes
+
+To keep configurations minimal, we will require that the certificates be mounted to "/tmp/k8s-metrics-server/metrics-certs" 
```

Comment:
Why tmp? Mounts rarely happen in that place.

Reply:
I mainly took the inspiration from the path we use for internal cert rotations.

https://github.com/kubernetes-sigs/kueue/blob/main/pkg/util/cert/cert.go#L31

Reply:
Right, but the internal certs rotations are temp and localized in principle. The external certs are expected to be "permanent", so indeed the "tmp" folder name is a bit misleading. Wondering if there are some best practices for it.

Reply:
I don't really see anything in cert docs.

We could put in /etc ?

Reply:
I pushed an update to put in /etc rather than /tmp. I think that is a better location for these certs.

Reply:
`/etc/kueue/certs` sgtm

Reply:
/etc/metrics/certs
or
/etc/kueue/metrics/certs?

We should not combine all the certificates into the same directory.

Reply:
>/etc/metrics/certs or /etc/kueue/metrics/certs?

Probably ` /etc/kueue/metrics/certs`, because in /etc the files are typically namespaces by the app name. Sure, we have only single app here, so everything works, but it feels more natural. OTOH, this is not a strong opinion.

> We should not combine all the certificates into the same directory.

Sure, if users want to use the same certs for webhooks and metrics they would need to create soft link between the two folders or just copy them to both places.

Reply:
PTAL. Updated.